### PR TITLE
fix: Treat webmanifest as an entry module

### LINKED ISSUE #1466: webmanifest name mangling breaks PWA updates
## 🤔 Expected Behavior

Updating the manifest.webmanifest for a PWA that was added to home screen should cause the application to update.

## 😯 Current Behavior

The application is not updated.

Background: on Android, PWAs can be installed along with native applications: https://developers.google.com/web/fundamentals/app-install-banners/ When the manifest is changed remotely, the PWA is repackaged and updated.

However, the way parcel renames the manifest whenever its content changes however breaks the update functionality. From Peter Kotwicz (WebAPK team):

> - We will only update a WebAPK if the Web Manifest URL does not change. If you change the web page from referencing <link rel="manifest.json"> to reference <link rel="manifest2.json"> the WebAPK will no longer update [(source)](https://groups.google.com/a/chromium.org/d/msg/chromium-discuss/NUjaM4QtFrU/gTJ1kSONBAAJ)

## 💁 Possible Solution

Add manifest.webmanifest to the exceptions outlined in #872 and implemented in #1025.

## 🔦 Context

I spent some time trying to get a PWA to update before researching reasons it might not update and stumbling over the newsgroup post quoted above.

### PR DESCRIPTION
**Pull request by AndreasPizsa**
treat webmanifest as an entry module so it doesn't get content hashed

closes #1466

# ↪️ Pull Request

Hashing the Webmanifest filename breaks PWA updates.
This PR makes webmanifest an entry module by default so it doesn't get content hashed.

## 💻 Examples

Bad:

```html
<html>
<head>
    <link rel="manifest" href="manifest.abcdef.webmanifest"> <!-- breaks auto update when hash changes -->
</head>
</html>
```

Good:
```html
<html>
<head>
    <link rel="manifest" href="manifest.webmanifest"> <!-- keeps auto update intact -->
</head>
</html>
```

## 🚨 Test instructions

Unit test included

## ✔️ PR Todo

- [x] Added/updated unit tests for this change
- [x] Filled out test instructions (In case there aren't any unit tests)
- [x] Included links to related issues/PRs

### CODE REVIEW

File: packages/core/parcel-bundler/src/assets/HTMLAsset.js [hunk 1]
```diff
@@ -163,6 +177,7 @@ class HTMLAsset extends Asset {
           if (elements && elements.includes(node.tag)) {
             let depHandler = this.getAttrDepHandler(attr);
             let options = OPTIONS[node.tag];
+
```
Comment (AndreasPizsa) [inline packages/core/parcel-bundler/src/assets/HTMLAsset.js]:
```suggestion
```

**Comment by AndreasPizsa** [comment]
Checks fail because of what look like unrelated issues to me, e.g.

```
2018-11-05T20:23:08.6972366Z   1) WorkerFarm
2018-11-05T20:23:08.6972641Z        Should start up workers:
2018-11-05T20:23:08.6972944Z      Error: Timeout of 2000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves. (/home/vsts/work/1/s/packages/core/workers/test/workerfarm.js)
2018-11-05T20:23:08.6973212Z       at process._tickCallback (internal/process/next_tick.js:109:7)
```

I'll happily fix if those if they are related, can you please have a look @devongovett ?

File: packages/core/workers/test/workerfarm.js [hunk 2]
```diff
@@ -2,6 +2,8 @@ const assert = require('assert');
 const WorkerFarm = require('../index');
 
 describe('WorkerFarm', () => {
+  this.timeout(10000);
```
Comment (DeMoorJasper) [inline packages/core/workers/test/workerfarm.js]: Probably better to change this for all workerfarm tests by changing `packages/core/workers/test/mocha.opts`

Comment (AndreasPizsa) [reply packages/core/workers/test/workerfarm.js]: True - I’m still learning how the code is organized.
Thanks for the feedback.

File: packages/core/workers/test/workerfarm.js [hunk 3]
```diff
@@ -1,7 +1,7 @@
 const assert = require('assert');
 const WorkerFarm = require('../index');
 
-describe('WorkerFarm', () => {
+describe('WorkerFarm', function() {
```
Comment (AndreasPizsa) [inline packages/core/workers/test/workerfarm.js]:
```suggestion
describe('WorkerFarm', () => {
```

**Comment by AndreasPizsa** [comment]
So this is now failing because of Azure Devops server error. Reminds me of how much I love Travis :)

I tried to re-run the build on Azure Devops, but it seems like I can't do that without pushing a new commit (maybe because of a lack of permissions?)

Any ideas how we can unblock this PR @DeMoorJasper ? Thanks!

**Comment by devongovett** [comment]
I've triggered a re-run.

**Comment by AndreasPizsa** [comment]
Can you please re-trigger again @devongovett? Thanks!

**Comment by DeMoorJasper** [comment]
@AndreasPizsa I've re-triggered the tests

**Pull request merged**